qa/tasks/cephfs/test_nfs.py

```diff
@@ -395,3 +395,75 @@ def test_cluster_info(self):
             }]}
         self.assertDictEqual(info_output, host_details)
         self._test_delete_cluster()
+
+    def test_cluster_set_reset_user_config(self):
+        '''
+        Test cluster is created using user config and reverts back to default
+        config on reset.
+        '''
+        config = """ LOG {
+        Default_log_level = FULL_DEBUG;
+        }
+
+        EXPORT {
+	        Export_Id = 100;
+	        Transports = TCP;
+	        Path = /;
+	        Pseudo = /ceph/;
+	        Protocols = 4;
+	        Access_Type = RW;
+	        Attr_Expiration_Time = 0;
+	        Squash = None;
+	        FSAL {
+	              Name = CEPH;
+	        }
+        }"""
+
+        self._test_create_cluster()
+        time.sleep(30)
+        #{'test': [{'hostname': 'smithi068', 'ip': ['172.21.15.68'], 'port': 2049}]}
+        info_output = json.loads(self._nfs_cmd('cluster', 'info', self.cluster_id))['test'][0]
+        ip = info_output['ip'][0]
+        port = info_output['port']
+        self._nfs_cmd('cluster', 'config', 'set', self.cluster_id, '-i', '-', stdin=config)
+        time.sleep(30)
+        mnt_cmd = ['mount', '-t', 'nfs', '-o', f'port={port}', f'{ip}:/ceph', '/mnt']
+        self._sys_cmd(mnt_cmd)
+        self._sys_cmd(['touch', '/mnt/test'])
+        opt_mnt = self._sys_cmd(['ls', '/mnt'])
+        log.info(opt_mnt)
+        self._sys_cmd(['umount', '/mnt'])
+        self._nfs_cmd('cluster', 'config', 'reset', self.cluster_id)
+        time.sleep(30)
+        self._sys_cmd(mnt_cmd)
+        self._test_delete_cluster()
+        time.sleep(30)
+        rados_obj_ls = self._sys_cmd(['rados', '-p', 'nfs-ganesha', '-N', self.cluster_id, 'ls'])
+        if b'conf-nfs' in rados_obj_ls and b'userconf-nfs' not in rados_obj_ls:
+            self.fail("User config not deleted")
```

Comment:
This check should go after `config reset`, right?

Reply:
Yes, I have modified the test.